src/simsopt/geo/curvealongz.py

```diff
@@ -0,0 +1,81 @@
+import jax.numpy as jnp
+from math import pi
+import numpy as np
+from .curve import JaxCurve
+
+__all__ = ['CurveAlongZ']
+
+
+def jaxcurvealongz_pure(dofs, quadpoints, zscale):
+    """
+    Pure function for the CurveAlongZ, returns the points along the curve given the degrees of freedom and the quadpoints. 
+
+    Args:
+        dofs: [xpos, ypos]; degrees of freedom of the curve, must be len(3)
+        quadpoints: points in [0,1]; points on which to evaluate the curve
+    """
+    lenquadpoints = len(quadpoints)
+    x = dofs[0]*jnp.ones(lenquadpoints)
+    y = dofs[1]*jnp.ones(lenquadpoints)
+    if lenquadpoints < 2:
+        halfstep = 0 # don't offset for one point or two points
+    else:
+        halfstep = .5/(lenquadpoints-1)  # avoid evaluating at 0 or 1. 
+    z = jnp.tan(((quadpoints + halfstep) - .5)*pi)*zscale
+    gamma = jnp.stack((x, y, z), axis=1)
+    return gamma
+
+
+class CurveAlongZ(JaxCurve):
+    r'''
+    Straight vertical curve, parallel to the z-axis. 
+
+    Useful for quickly generating a toroidal field, comparing to tokamak equilibria
+    where an axisymmetric 1/R field is present, and testing. 
```

Comment:
do you really want this approximation of an axisymmetric 1/R field? why not just implement the analytical magnetic field instead of this approximation?